walnut/converter/converter.go

```diff
@@ -102,48 +80,89 @@ func ExtractTaskClasses(dplDump Dump) (tasks []*task.Class, err error) {
 			Command: &common.CommandInfo{
 				Env:       []string{}, // -> Default to empty array
 				Shell:     createBool(true),
-				Value:     &dplDump.Metadata[correspondingMetadata].Executable,
-				Arguments: dplDump.Metadata[correspondingMetadata].CmdlLineArgs,
+				Arguments: sanitizeCmdLineArgs(dplDump.Metadata[correspondingMetadata].CmdlLineArgs,
+					taskName),
 				User:      createString("flp"),
 			},
 			Wants: task.ResourceWants{
 				Cpu:    createFloat(0.15),
 				Memory: createFloat(128),
-				Ports:  task.Ranges{}, //begin - end OR range
+				Ports:  task.Ranges{}, // begin - end OR range
 			},
-			Bind: bind,
 			Properties: gera.MakeStringMapWithMap(map[string]string{
 				"severity": "trace",
 				"color":    "false",
 			}),
-			Connect: connect,
 		}
-		// fmt.Printf("\nTASK:\n%v\n", task)
+
+		value := fmt.Sprintf("eval `aliswmod load %s` &&\n%s", strings.Join(envModules, " "),
+			dplDump.Metadata[correspondingMetadata].Executable)
+		task.Command.Value = &value
+
+
+		for _, channelName := range channelNames {
+			// To avoid duplication, only "push" channels are included
+			if strings.Contains(channelName, "from_"+taskName) {
+				singleBind := channel.Inbound{
+					Channel: channel.Channel{
+						Name:      channelName,
+						Type:      channel.ChannelType("push"),
+						Transport: channel.TransportType("shmem"),
+					},
+					Addressing: "ipc",
+				}
+				task.Bind = append(task.Bind, singleBind)
+			}
+
+			if strings.Contains(channelName, "to_"+taskName) {
+				// String manipulation to generate channel target of the form:
+				// {{ Parent().Path }}.taskName:from_{initiator}_to_{receiver}
+				initiator := channelName[5:strings.Index(channelName, "_to")]
+				singleConnect := channel.Outbound{
+					Channel: channel.Channel{
+						Name:      channelName,
+						Type:      channel.ChannelType("pull"),
+						Transport: channel.TransportType("shmem"),
+					},
+					Target: fmt.Sprintf("{{ Parent().Path }}.%s:%s", initiator, channelName),
+				}
+				task.Connect = append(task.Connect, singleConnect)
+			}
+		}
+
 		tasks = append(tasks, &task)
 	}
 	return tasks, nil
 }
 
-// TaskToYAML takes as input an array of pointers to task.Class
-// and writes them to a AliECS friendly YAML file
-func TaskToYAML(extractedTasks []*task.Class) (err error) {
-
-	// Check if "tasks" directory exists. If not, create it
-	_, err = os.Stat("tasks")
-	if os.IsNotExist(err) {
-		// FIXME: use absolute paths
-		errDir := os.Mkdir("tasks", 0755)
-		if errDir != nil {
-			return fmt.Errorf("create dir failed: %v", err)
+func sanitizeCmdLineArgs (input []string, taskName string) (output []string) {
+	for _, value := range input {
+		// Check args for dump arguments and remove them
+		if  strings.Contains(value, "--dump-workflow") ||
+			strings.Contains(value, "--dump-workflow-file") ||
+			strings.Contains(value, ".json") {} else {
```

Comment:
Might not be safe to remove all params that end with `.json`.

Reply:
I'll remove the check for `.json` but in that case we will have the name of the dump (to which it was exported to) passed onto `cmdLineArgs`.

Reply:
Not necessarily. You can only remove the first item after `--dump-workflow-file` if this item is `.json`, since you know the input slice is ordered.